### PR TITLE
chore: move to `async-graphql-parser`

### PR DESCRIPTION
**Pull request by deekerno**
## Changelog
- Move packages from use of `graphql-parser`/`fuel-indexer-graphql-parser` to `async-graphql-parser`
- Remove `fuel-indexer-graphql-parser` package
- Adjust tests

## Testing Plan
CI should pass.

### Manual Testing
Feel free to run the block explorer example with the simple and complex example queries from the documentation.

## Notes
This is the first step in integrating `async-graphql` into the indexer.

### CODE REVIEW

File: packages/fuel-indexer-graphql/src/arguments.rs [hunk 1]
```diff
@@ -438,15 +438,15 @@ pub fn parse_argument_into_param<'a>(
...
             } else {
                 Err(GraphqlError::UnsupportedValueType(value.to_string()))
             }
         }
         "first" => {
-            if let Value::Int(limit) = value {
-                Ok(ParamType::Limit(limit.as_u64()))
+            if let Value::Number(limit) = value {
+                Ok(ParamType::Limit(limit.as_u64().unwrap()))
             } else {
```
Comment (rvmelkonian) [inline packages/fuel-indexer-graphql/src/arguments.rs:450]: Same question as above

File: packages/fuel-indexer-graphql/src/arguments.rs [hunk 2]
```diff
@@ -721,11 +721,11 @@ fn parse_binary_logical_operator<'a>(
...
 /// Value types from the parsed GraphQL query should be turned into `ParsedValue`
 /// instances so that they can be properly formatted for transformation into SQL queries.
-fn parse_value<'a>(value: &Value<'a, &'a str>) -> Result<ParsedValue, GraphqlError> {
+fn parse_value(value: &Value) -> Result<ParsedValue, GraphqlError> {
     match value {
-        Value::BigInt(bn) => Ok(ParsedValue::BigNumber(bn.as_u128())),
+        // TODO: Add support for u128 through the use of a custom scalar
         Value::Boolean(b) => Ok(ParsedValue::Boolean(*b)),
-        Value::Int(n) => Ok(ParsedValue::Number(n.as_u64())),
+        Value::Number(n) => Ok(ParsedValue::Number(n.as_u64().unwrap())),
```
Comment (rvmelkonian) [inline packages/fuel-indexer-graphql/src/arguments.rs:728]: If this can fail, do you want the program to panic?

**Review by rvmelkonian — CHANGES_REQUESTED** [review]
Great stuff with this. Left a few small comments, after those I will test this and try to break with the playground.

**Comment by deekerno** [comment]
@rvmelkonian Oops, I thought I got all of my testing `.unwrap()`s removed. 😅 I'll remove and ping again.

**Review by ra0x3 — APPROVED** [review]
- Hello-world example works ✅ 
- Don't see any obvious issues
- Don't want to leave this one hanging on the vine for too long

**Comment by rvmelkonian** [comment]
@deekerno e2e tests and unit test are failing at the master rebase.

**Comment by deekerno** [comment]
I love fixing things that I've already fixed.

**Review by rvmelkonian — APPROVED** [review]

**Pull request merged**